packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,497 @@
+    super.mount(parent, newSlot);
+    for (int i = 0; i < widget.children.length; i += 1) {
+      _mountChild(widget.children.elementAt(i), i);
+    }
+  }
+
+  void _mountChild(Widget widget, int slot) {
+    Element oldChild;
+    if (slot >= slots.length) {
+      oldChild = null;
+    } else {
+      oldChild = slots.elementAt(slot);
+    }
+    final Element newChild = updateChild(oldChild, widget, slot);
+    if (oldChild != null) {
+      slots.remove(oldChild);
+    }
+    if (newChild != null) {
+      slots.insert(slot, newChild);
+    }
+  }
+}
+
+class _RenderSegmentedControl extends RenderBox {
+  _RenderSegmentedControl(
+      {@required this.children, @required this.selectedValue, @required this.pressedValue});
+
+  List<Widget> children;
+
+  Key selectedValue, pressedValue;
+
+  static BoxParentData _boxParentData(RenderBox box) => box.parentData;
+
+  static Size _boxSize(RenderBox box) => box == null ? Size.zero : box.size;
+
+  static Rect _boxRect(RenderBox box) =>
+      box == null ? Rect.zero : _boxParentData(box).offset & box.size;
+
+  static double _minWidth(RenderBox box, double height) {
+    return box == null ? 0.0 : box.getMinIntrinsicWidth(height);
+  }
+
+  static double _minHeight(RenderBox box, double width) {
+    return box == null ? 0.0 : box.getMinIntrinsicHeight(width);
+  }
+
+  final List<RenderBox> slots = <RenderBox>[];
+
+  final List<RenderBox> _boxes = <RenderBox>[];
+
+  List<RRect> rects = <RRect>[];
+
+  List<RenderBox> get boxes => _boxes;
+
+  RenderBox getBox(int slot) {
+    return boxes.elementAt(slot);
+  }
+
+  void setBox(int slot, RenderBox value) {
+    if (slot >= slots.length) {
+      boxes.insert(slot, _updateChild(null, value, slot));
+    } else {
+      boxes.insert(slot, _updateChild(boxes.elementAt(slot), value, slot));
+    }
+  }
+
+  RenderBox _updateChild(RenderBox oldChild, RenderBox newChild, int slot) {
+    if (oldChild != null) {
+      dropChild(oldChild);
+      slots.remove(oldChild);
+    }
+    if (newChild != null) {
+      slots.insert(slot, newChild);
+      adoptChild(newChild);
+    }
+    return newChild;
+  }
+
+  Iterable<RenderBox> get _children sync* {
+    for (int i = 0; i < boxes.length; i += 1) {
+      if (boxes.elementAt(i) != null) {
+        yield boxes.elementAt(i);
+      }
+    }
+  }
+
+  @override
+  void attach(PipelineOwner owner) {
+    super.attach(owner);
+    for (RenderBox child in _children) {
+      child.attach(owner);
+    }
+  }
+
+  @override
+  void detach() {
+    super.detach();
+    for (RenderBox child in _children) {
+      child.detach();
+    }
+  }
+
+  @override
+  void redepthChildren() {
+    _children.forEach(redepthChild);
+  }
+
+  @override
+  void visitChildren(RenderObjectVisitor visitor) {
+    _children.forEach(visitor);
+  }
+
+  @override
+  double computeMinIntrinsicWidth(double height) {
+    double total;
+    for (int i = 0; i < boxes.length; i += 1) {
```

Comment:
This should be the other kind of `for` loop (because it's more readable):
```dart
for (RenderBox box in boxes) {
  total += _minWidth(box, height);
}
```